bayesian/api_v1.py

```diff
@@ -275,46 +275,41 @@ def get(ecosystem, package, version):
 
 class StackAnalysesByGraphGET(ResourceWithSchema):
     method_decorators = [login_required]
-
     schema_ref = SchemaRef('stack_analyses', '2-1-4')
 
     @staticmethod
     def get(external_request_id):
-        try:
-            results = rdb.session.query(WorkerResult)\
-                                 .filter(WorkerResult.external_request_id == external_request_id,
-                                         or_(WorkerResult.worker == "stack_aggregator",
-                                             WorkerResult.worker == "recommendation"))
-            if results.count() <= 0:
-                raise HTTPError(202, "Analysis for request ID '{t}' is in progress".format(t=external_request_id))
-        except SQLAlchemyError:
+        stack_result = retrieve_worker_result(rdb, external_request_id, "stack_aggregator")
+        reco_result = retrieve_worker_result(rdb, external_request_id, "recommendation")
+
+        if stack_result == None and reco_result == None:
```

Comment:
Not really a problem, but `if stack_result is None ...` would be more pythonic :)